Apps/Examples/Examples/All Examples/SnapshotterExample.swift

```diff
@@ -8,65 +8,79 @@ public class SnapshotterExample: UIViewController, ExampleProtocol {
                                                                          zoom: 3.5),
                                             styleURI: .dark)
 
         mapView = MapView(frame: testRect, mapInitOptions: mapInitOptions)
-        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
-
         // Add the `MapViewController`'s view to the stack view as a
         // child view controller.
         stackView.addArrangedSubview(mapView)
 
         // Add the image view to the stack view, which will eventually contain the snapshot.
-        snapshotView = UIImageView(frame: CGRect.zero)
+        let stackViewBounds = CGRect(x: 0,
+                                     y: 0,
+                                     width: view.bounds.size.width,
+                                     height: view.bounds.height / 2)
+        snapshotView = UIImageView(frame: stackViewBounds)
         stackView.addArrangedSubview(snapshotView)
 
+        NSLayoutConstraint.activate([
+            mapView.widthAnchor.constraint(equalToConstant: view.bounds.size.width),
+            snapshotView.widthAnchor.constraint(equalToConstant: stackViewBounds.width),
+            snapshotView.heightAnchor.constraint(equalToConstant: stackViewBounds.height)
+        ])
+
         // Add the stack view to the root view.
         view.addSubview(stackView)
 
         // Configure the snapshotter object with its default access
         // token, size, map style, and camera.
         let options = MapSnapshotOptions(
-            size: CGSize(
-                width: view.bounds.size.width,
-                height: view.bounds.height / 2),
+            size: stackViewBounds.size,
             pixelRatio: UIScreen.main.scale,
             resourceOptions: mapInitOptions.resourceOptions)
 
         snapshotter = Snapshotter(options: options)
         snapshotter.style.uri = .light
 
         // Set the camera of the snapshotter
-        let snapshotterCameraOptions = CameraOptions(cameraState: mapView.cameraState)
-        snapshotter.setCamera(to: snapshotterCameraOptions)
 
-        snapshotter.onNext(.styleLoaded) { [weak self] _ in
-            self?.startSnapshot()
+        mapView.mapboxMap.onEvery(.mapIdle) { [weak self] _ in
+            guard let self = self, !self.snapshotting else {
+                return
+            }
+
+            let snapshotterCameraOptions = CameraOptions(cameraState: self.mapView.cameraState)
+            self.snapshotter.setCamera(to: snapshotterCameraOptions)
+            self.startSnapshot()
         }
     }
 
     public func startSnapshot() {
+        snapshotting = true
         snapshotter.start(overlayHandler: nil) { ( result ) in
             switch result {
             case .success(let image):
                 self.snapshotView.image = image
             case .failure(let error):
                 print("Error generating snapshot: \(error)")
             }
+            self.snapshotting = false
```

Comment:
this snapshotting property seems useless. I know you use it up above in the map idle guard check, but it can be removed

Reply:
This is so we don't start a new snapshot while another is in progress. This example changed to take a new snapshot on each map idle. FYI @jmkiley

Reply:
Discussed in person.